base/log_keys.go

```diff
@@ -38,132 +41,155 @@ const (
 	KeySyncMsg
 	KeyWebSocket
 	KeyWebSocketFrame
-)
 
-var (
-	logKeyNames = map[LogKey]string{
-		KeyNone:           "",
-		KeyAll:            "*",
-		KeyAdmin:          "Admin",
-		KeyAccess:         "Access",
-		KeyAuth:           "Auth",
-		KeyBucket:         "Bucket",
-		KeyCache:          "Cache",
-		KeyChanges:        "Changes",
-		KeyCRUD:           "CRUD",
-		KeyDCP:            "DCP",
-		KeyEvents:         "Events",
-		KeyGoCB:           "gocb",
-		KeyHTTP:           "HTTP",
-		KeyHTTPResp:       "HTTP+", // Infof printed as HTTP+
-		KeyImport:         "Import",
-		KeyJavascript:     "Javascript",
-		KeyMigrate:        "Migrate",
-		KeyQuery:          "Query",
-		KeyReplicate:      "Replicate",
-		KeySync:           "Sync",
-		KeySyncMsg:        "SyncMsg",
-		KeyWebSocket:      "WS",
-		KeyWebSocketFrame: "WSFrame",
-	}
+	LogKeyCount // Count for logKeyNames init
+)
 
-	// Inverse of the map above. Optimisation for string -> LogKey lookups in ToLogKey
+var logKeyNames []string
```

Comment:
We can make this an array instead of a slice, and initialize outside of the init to provide some compile-time checking for values (e.g. If we include an extra value somewhere).

```suggestion
var logKeyNames [LogKeyCount]string{
	KeyNone:     "",
	KeyAll:      "*",
	// ...
	23: "extra",
}
```

`base/log_keys.go:72:2: array index 23 out of bounds [0:23]`

Reply:
Good idea - will change